### PR TITLE
Wit.ai Speech to Text for Question Answering

### PR DESCRIPTION
**Pull request by tuzhucheng**
Add a toolbar icon - when clicked, it will open a small window that allows user to record audio. The audio is recorded in WAV format and sent to the wit.ai speech API, which converts speech to text. Once the text is returned we call the Flask `/answer` endpoint to get a result for the query.

image

### CODE REVIEW

File: src/main/js/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules/
+SpeechDemo-darwin-x64/
```
Comment (lintool) [inline src/main/js/.gitignore]: shouldn't this be in the top-level .gitignore?

File: src/main/js/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+npm start
+```
+
+You should see a new icon in the toolbar. Click it and a small window will open. Use the keyboard shortcut for Chrome developer tools to open up the console. You'll need to type in the following command to enter the API key before using it:
+
+```javascript
+window.WITAI_API_SECRET = "Bearer XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
+```
+
+Ask me for the API key.
```
Comment (lintool) [inline src/main/js/README.md]: I think by convention we're throwing all the documentation in `doc/`

**Comment by salman1993** [comment]
looks good to me!

File: src/main/js/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Setting up Speech to Text Demo
+
+```sh
```
Comment (salman1993) [inline src/main/js/README.md]: add  src/main/js
it's trivial but some people might miss it

Comment (tuzhucheng) [inline src/main/js/README.md]: Ah after I moved it to docs I updated this already.

Comment (salman1993) [inline src/main/js/README.md]: great! 👍

Comment (tuzhucheng) [inline src/main/js/README.md]: Actually this is not just exclusively for OS X. I'm using, https://github.com/maxogden/menubar, can run on Mac OS X, Windows, and some desktop environments on Linux (e.g. tested on xfce) at least.

Comment (salman1993) [inline src/main/js/README.md]: do you know why Electron was failing on my Ubuntu machine?

Comment (tuzhucheng) [inline src/main/js/README.md]: I assume you are using Unity (default desktop environment for Ubuntu), which is probably not supported (yet).

**Pull request merged**